shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/context/CommonBeans.java

```diff
@@ -72,10 +73,14 @@
   private static final Logger LOG = LoggerFactory.getLogger(CommonBeans.class);
 
   @Bean
-  Configuration baseConfiguration(@Value("${instance.workspace}") String workspace) {
+  Configuration baseConfiguration(
+      @Value("${instance.workspace}") String workspace,
+      @Value("${ct-config}") String circusTrainConfigLocation) {
     checkNotNull(workspace, "instance.workspace is required");
+    checkNotNull(circusTrainConfigLocation, "ct-config is required");
```

Comment:
Being pedantic here since we're the only users of Shunting Yard up to now, but this would be a backwards incompatible change for end users right? We should always think carefully about these things, if we really want to do them that's fine they just require a major version bump and a clear line in the CHANGELOG describing the change and what users need to do. 

Could there be a situation where users don't want to provide a ct config so we could keep this optional?

Reply:
I think this doesn't have to be required. The basic config is made on the fly, this "common" one contains extras (grafana, maybe sns). So can be optional.

Reply:
The argument is optional now.